packages/patrol/example/integration_test/contacts_test.dart

```diff
@@ -0,0 +1,163 @@
+    await $.native.pressHome();
+    await $.native.openApp(appId: contactsAppId);
+
+    final first = await _createNewContact(
+      $,
+      firstName: 'Charlie',
+      lastName: 'Root',
+    );
+    final second = await _createNewContact(
+      $,
+      firstName: 'Charlie',
+      lastName: 'Root',
+    );
+
+    await $.native.tap(
+      Selector(text: first, instance: 0),
+      appId: contactsAppId,
+    );
+    // TODO: Verify that first contact is shown
+
+    await _goBack($);
+
+    await $.native.tap(
+      Selector(text: second, instance: 1),
+      appId: contactsAppId,
+    );
+    // TODO: Verify that second contact is shown
+
+    await _goBack($);
+  });
+}
+
+/// Creates a new contact with random data.
+///
+/// Returns the first name and last name of the created contact.
+Future<String> _createNewContact(
+  PatrolTester $, {
+  String? firstName,
+  String? lastName,
+}) async {
+  if (Platform.isIOS) {
+    return _createContactOnIOS($, firstName: firstName, lastName: lastName);
+  } else if (Platform.isAndroid) {
+    return _createContactOnAndroid($, firstName: firstName, lastName: lastName);
+  }
+
+  throw UnsupportedError('unsupported platform');
+}
+
+Future<String> _createContactOnIOS(
+  PatrolTester $, {
+  String? firstName,
+  String? lastName,
+}) async {
+  firstName = firstName ?? faker.person.firstName();
+  lastName = lastName ?? faker.person.lastName();
+  final company = faker.company.name();
+  final phoneNumber = faker.phoneNumber.us();
+  final email = faker.internet.email();
+
+  await $.native.tap(Selector(text: 'Add'), appId: contactsAppId);
+  await $.native.enterTextByIndex(firstName, index: 0, appId: contactsAppId);
+  await $.native.enterTextByIndex(lastName, index: 1, appId: contactsAppId);
+  await $.native.enterTextByIndex(company, index: 2, appId: contactsAppId);
+
+  await $.native.tap(Selector(text: 'add phone'), appId: contactsAppId);
+  await $.native.enterText(
+    Selector(text: 'Phone'),
+    text: phoneNumber,
+    appId: contactsAppId,
+  );
+
+  await $.native.tap(Selector(text: 'add email'), appId: contactsAppId);
+  await $.native.enterText(
+    Selector(text: 'Email'),
+    text: email,
+    appId: contactsAppId,
+  );
+
+  await $.native.tap(Selector(text: 'Done'), appId: contactsAppId);
+
+  await _goBack($);
+
+  return '$firstName $lastName';
+}
+
+Future<String> _createContactOnAndroid(
+  PatrolTester $, {
+  String? firstName,
+  String? lastName,
+}) async {
+  firstName = firstName ?? faker.person.firstName();
+  lastName = lastName ?? faker.person.lastName();
+  final company = faker.company.name();
+  final phoneNumber = faker.phoneNumber.us();
+  final email = faker.internet.email();
+
+  await $.native.tap(Selector(contentDescription: 'Create contact'));
+  await $.native.enterTextByIndex(firstName, index: 0);
+  await $.native.enterTextByIndex(lastName, index: 1);
+  await $.native.enterTextByIndex(company, index: 2);
```

Comment:
This `enterTextByIndex()` can vary from device to device. Not all android device's `index: 2` has the `company` field.

Reply:
Unfortunately, it's not possible to make this test work on all `Contacts app` that exist.